### PR TITLE
Update to Vulkan 1.0.49 header

### PR DESCRIPTION
**Pull request by mark-lunarg**
New scripts, header files, new Vulkan.hpp including a workaround for getSurfaceCapabilities2KHR issues, parameter_validation, unique_objects and object_tracker layer support.

### CODE REVIEW

**Comment by tobine** [comment]
If you let my #1767 land first and rebase on that, then you'll have IDs for some of the UNDEFINED cases. I'll go through the new checks in this PR and note new IDs.

File: layers/object_tracker.cpp [hunk 1]
```diff
@@ -5041,6 +5041,61 @@ VKAPI_ATTR void VKAPI_CALL SetHdrMetadataEXT(VkDevice device, uint32_t swapchain
     }
 }
 
+VKAPI_ATTR VkResult VKAPI_CALL GetSwapchainStatusKHR(VkDevice device, VkSwapchainKHR swapchain) {
+    bool skip = false;
+    {
+        std::lock_guard<std::mutex> lock(global_lock);
+        skip |=
+            ValidateObject(device, device, kVulkanObjectTypeDevice, false, VALIDATION_ERROR_UNDEFINED, VALIDATION_ERROR_UNDEFINED);
```
Comment (tobine) [inline layers/object_tracker.cpp]: VALIDATION_ERROR_03293

File: layers/object_tracker.cpp [hunk 2]
```diff
@@ -5041,6 +5041,61 @@ VKAPI_ATTR void VKAPI_CALL SetHdrMetadataEXT(VkDevice device, uint32_t swapchain
...
+        skip |= ValidateObject(device, swapchain, kVulkanObjectTypeSwapchainKHR, false, VALIDATION_ERROR_UNDEFINED,
```
Comment (tobine) [inline layers/object_tracker.cpp]: VALIDATION_ERROR_03294

File: layers/object_tracker.cpp [hunk 3]
```diff
@@ -5041,6 +5041,61 @@ VKAPI_ATTR void VKAPI_CALL SetHdrMetadataEXT(VkDevice device, uint32_t swapchain
...
+    return result;
+}
+
+VKAPI_ATTR VkResult VKAPI_CALL GetPhysicalDeviceSurfaceCapabilities2KHR(VkPhysicalDevice physicalDevice,
+                                                                        const VkPhysicalDeviceSurfaceInfo2KHR *pSurfaceInfo,
+                                                                        VkSurfaceCapabilities2KHR *pSurfaceCapabilities) {
+    bool skip = false;
+    {
+        std::lock_guard<std::mutex> lock(global_lock);
+        skip |= ValidateObject(physicalDevice, physicalDevice, kVulkanObjectTypePhysicalDevice, false, VALIDATION_ERROR_UNDEFINED,
```
Comment (tobine) [inline layers/object_tracker.cpp]: VALIDATION_ERROR_03283

File: layers/object_tracker.cpp [hunk 4]
```diff
@@ -5041,6 +5041,61 @@ VKAPI_ATTR void VKAPI_CALL SetHdrMetadataEXT(VkDevice device, uint32_t swapchain
...
+                               VALIDATION_ERROR_UNDEFINED);
+        skip |= ValidateObject(physicalDevice, pSurfaceInfo->surface, kVulkanObjectTypeSurfaceKHR, false,
```
Comment (tobine) [inline layers/object_tracker.cpp]: VALIDATION_ERROR_03288

File: layers/object_tracker.cpp [hunk 5]
```diff
@@ -5041,6 +5041,61 @@ VKAPI_ATTR void VKAPI_CALL SetHdrMetadataEXT(VkDevice device, uint32_t swapchain
...
+}
+
+VKAPI_ATTR VkResult VKAPI_CALL GetPhysicalDeviceSurfaceFormats2KHR(VkPhysicalDevice physicalDevice,
+                                                                   const VkPhysicalDeviceSurfaceInfo2KHR *pSurfaceInfo,
+                                                                   uint32_t *pSurfaceFormatCount,
+                                                                   VkSurfaceFormat2KHR *pSurfaceFormats) {
+    bool skip = false;
+    {
+        std::lock_guard<std::mutex> lock(global_lock);
+        skip |= ValidateObject(physicalDevice, physicalDevice, kVulkanObjectTypePhysicalDevice, false, VALIDATION_ERROR_UNDEFINED,
```
Comment (tobine) [inline layers/object_tracker.cpp]: VALIDATION_ERROR_03289

**Comment by mark-lunarg** [comment]
@tobine, I have a commit with these changes.  I would appreciate if you could verify the VUID for vkGetPhysicalDeviceSurfaceFormats2KHR's pSurfaceInfo->surface -- is it the same as the one for GetPhysicalDeviceSurfaceCapabilities2KHR's pSurfaceInfo->surface?

**Comment by mark-lunarg** [comment]
Had to fix an unexpected error in the LVTs -> InvalidImageLayouts was trying to match the part of the string from the spec, which changed in the header update.

**Pull request merged**